modules/slider/Slider.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect } from "react";
+import { withStyles } from "@diana-ui/base";
+import { ThemeStyleSheetFactory, WithStylesProps } from "@diana-ui/types";
+import { LabelRegular } from "@diana-ui/typography";
+
+export interface ISliderProps {
+  min: number;
+  max: number;
+  value: number;
+  step: number;
+  onChange: any;
+  disabled?: boolean;
+  className?: string;
+}
+
+export interface ISliderState {
+  leftSpacing: number;
+}
+
+const styleSheet: ThemeStyleSheetFactory = theme => ({
+  wrapper: {},
+  valueWrapper: {},
+  input: {
+    "&::-webkit-slider-thumb": {},
+    "@selectors": {
+      "&.disabled::-webkit-slider-thumb": {},
+      "&.disabled::-webkit-slider-runnable-track": {}
+    },
+    "&::-moz-range-thumb": {},
+    "&::-ms-thumb": {},
+    "&:focus": {},
+    "&::-ms-track": {},
+    "&::-webkit-slider-runnable-track": {}
+  },
+  value: {}
+});
+
+const calculateLeftSpace = (
```

Comment:
If you define this function inside a `useCallback` in the component, you can simplify the function signature by removing most parameters 💭 although I do understand why you defined it outside! Just a consideration

Reply:
Also, this formula is a bit hard to understand, at least at first glance. Do you think adding a comment describing what and how this function works would be helpful, or do you believe it is already as self-descriptive as possible?

Reply:
I would also add it inside as a useCallback. It would be easier to understand and it would remove params complexity